src/System.Private.ServiceModel/tools/scripts/OpenFirewallPorts.cmd

```diff
@@ -0,0 +1,18 @@
+@echo off 
+setlocal
+
+REM 
+REM Start Open all ports WCF test services use
+REM
+
+netsh advfirewall firewall add rule name="_WCF Test Server PortHttp" dir=in action=allow profile=any localport=8081 protocol=tcp
+netsh advfirewall firewall add rule name="_WCF Test Server PortHttps" dir=in action=allow profile=any localport=44285 protocol=tcp
+netsh advfirewall firewall add rule name="_WCF Test Server PortTcp" dir=in action=allow profile=any localport=809 protocol=tcp
+netsh advfirewall firewall add rule name="_WCF Test Server PortWebSocket" dir=in action=allow profile=any localport=8083 protocol=tcp
+netsh advfirewall firewall add rule name="_WCF Test Server PortWebSockets" dir=in action=allow profile=any localport=8084 protocol=tcp
+
+set __EXITCODE=%ERRORLEVEL%
```

Comment:
This _EXITCODE will only be for the final netsh command - that is, we're not testing the errorlevel for any of the previous commands, so we won't know what failed, and we're not bailing early if any of the previous rules are unsuccessful. 

Making matters slightly worse, someone who now looks in their firewall rules will see some of the rules created but not others,  probably making an assumption that all the rules were created just fine. 

I would prefer an all or nothing approach, which either means that we clean up all of the ones we tried to create if unsuccessful. At the very least we should report on them and note that _any_ of them failing will cause the script to bail out